### DISCUSSION TITLE
Probably misunderstanding how feed filters work

### DISCUSSION THREAD

**Discussion by pdxmph** [Athou/commafeed#1304] [General]
**Describe the bug**
Using the `url.contains` filter doesn't seem to work, but its inverse does.  I am guessing I am missing how this is supposed to work. 

**To Reproduce**

1. Edit a feed
2. Set up a filter to capture URLs with `advice` in them (`url.contains('advice')`)
3. Refresh the feed. 
4. Nothing is marked as read, including things that have `advice` in the URL. 
5. Revisit the filter and negate it with `not `url.contains('advice')`
6. Refresh. Everything but advice columns is marked read. 

**Expected behavior**
`url.contains('advice')` will mark anything with advice in the URL as read. Use case: My local paper puts all the advice columns under `https://oregonlive.com/advice/foo`. 

**Environment (please complete the following information):**

- CommaFeed version (or "commafeed.com"): 4.3.1 (fafd4c9)
- Browser [e.g. chrome, firefox]: Firefox
- Device [e.g. desktop, mobile]: Desktop (Linux, Fedora 39)

**Additional context**

**Comment by Athou** [comment]
Hi @pdxmph, 

> 3\. Refresh the feed.

The filter only applies to new entries, changing the filter and refreshing the feed will not mark existing entries as read.

> Refresh. Everything but advice columns is marked read.

I'm not sure how that's possible :D

There's an issue here to apply the filter to existing entries https://github.com/Athou/commafeed/issues/728

**Comment by dstutz** [comment]
I don't know if this belongs here, but my general question is basically the same one.  I don't seem to be able to get filtering to do anything for me.  I'm sure I'm doing something stupidly wrong.

I'm running 7.0.0 which has the new style filtering.  I understand that filtering only applies to new entries.

Take for example this feed: https://www.thebaltimorebanner.com/arc/outboundfeeds/rss-general/

I don't really care about the sports so I'm trying to mark as read stuff with "orioles" or "ravens" or "nationals".  What should the filter expression look like for that to happen?  One example should be enough.

This:
<img width="911" height="186" alt="image" src="https://github.com/user-attachments/assets/d413e529-bdf9-4867-a8ef-0cfcac6ae237" />
Is not working as I am still seeing new stuff come in with orioles in the title.

**Reply by omer-faruq** [reply]
You should switch to "AND" if you are using "not contain".

**Reply by dstutz** [reply]
Welp, that appears to be working now, thanks for helping my lack of brain.
